addons/website/models/website.py

```diff
@@ -722,7 +722,8 @@ def get_unique_path(self, page_url):
         # we only want a unique_path for website specific.
         # we need to be able to have /url for website=False, and /url for website=1
         # in case of duplicate, page manager will allow you to manage this case
-        domain_static = [('website_id', '=', self.get_current_website().id)]  # .website_domain()
+        website_id = self.env.context.get('website_id', False) or self.get_current_website().id
+        domain_static = [('website_id', '=', website_id)]  # .website_domain()
```

Comment:
I think it won't work this way since `get_current_website` can return the session website before [here](https://github.com/odoo-dev/odoo/blob/44dba0dab181659aceedf72a12db2bbdadada135/addons/website/models/website.py#L937) [**]

Reply:
Is that an issue?
Are you able to have a force_website_id let's say set to 2 and a website_id set to 1 in context?
Eg, from one of the caller:
```py
copy_param = dict(name=page_name or page.name, website_id=self.env['website'].get_current_website().id)
if page_name:
    copy_param['url'] = self.get_valid_page_url(page_name)
```

It already use the `get_current_website()`, without considering the context before the session force.